pandas/core/groupby/grouper.py

```diff
@@ -875,7 +875,7 @@ def is_in_obj(gpr) -> bool:
             exclusions.add(gpr.name)
 
         elif is_in_axis(gpr):  # df.groupby('name')
-            if gpr in obj:
+            if not isinstance(obj, Series) and gpr in obj:
```

Comment:
Maybe this should be `grp in group_axis`?

I am not 100% familiar with this code path but there may be a case where this can apply to the Series index?

Reply:
For Series index the if-statement one line before "is_in_axis(gpr)" gives true. But the if-statement "gpr in obj" is only for dataframes to check whether the group key gpr is one of the columns. For series (and gpr in series index) the corresponding else-statement is correct. The new test cases prove that the result is correct, if the group key is the series index. I checked that it is working with series that has MultiIndex as well.

Reply:
Agreed - this looks good to me. For a Series, we only need to consider the case `axis=0`. So when you have an "in-axis grouper" for a Series, I think there are conceptually only two options: the index, or the values itself. I believe grouping the by the values via the in-axis path, e.g.

```
ser = pd.Series([1, 2, 1, 2], index=[1, 1, 2, 2], name="b")
ser.index.name = "a"
ser.groupby("b").sum()
```

is entirely not supported (though `ser.groupby(ser).sum()` is supported - but that's a different code path) and is outside the scope here. So that only leaves checking if the provided grouper matches the index name, which is the elif block here.